menpo/io/input/video.py

```diff
@@ -59,53 +47,285 @@ def imageio_ffmpeg_importer(filepath, asset=None, normalize=True, **kwargs):
         A :map:`LazyList` containing :map:`Image` or subclasses per frame
         of the video.
     """
-    import imageio
-
-    reader = imageio.get_reader(str(filepath), format='ffmpeg', mode='I')
-
-    index_callable = partial(_imageio_to_menpo, reader, normalize)
-    ll = LazyList.init_from_index_callable(index_callable,
-                                           reader.get_length())
-    ll.fps = reader.get_meta_data()['fps']
-
-    if len(ll) != 0:
-        # TODO: Remove when imageio fixes the ffmpeg importer duration/start
-        # This is a bit grim but the frame->timestamp logic in imageio at
-        # the moment is not very accurate and so we really need to ensure
-        # that the user is returned a list they can actually index into. So
-        # we just remove all the frames that we can't actually index into.
-        # Remove from the front
-        for start in range(len(ll)):
-            if start > 10:  # Arbitrary but probably safe
-                warnings.warn('Highly inaccurate frame->timestamp mapping '
-                              'returned by imageio - many frames are being '
-                              'dropped and thus importing may be very slow.'
-                              ' Please see the documentation.')
-            try:
-                ll[start]
-                break
-            except:
-                pass
-        else:
-            # If we never broke out then ALL frames raised exceptions
-            ll = LazyList([])
-        # Only take the frames after the initial broken ones
-        ll = ll[start:]
-
-    if len(ll) != 0:
-        n_frames = len(ll) - 1
-        for end in range(n_frames, -1, -1):
-            if end < n_frames - 10:  # Arbitrary but probably safe
-                warnings.warn('Highly inaccurate frame->timestamp mapping '
-                              'returned by imageio - many frames are being '
-                              'dropped and thus importing may be very slow.'
-                              ' Please see the documentation.')
-            try:
-                ll[end]
-                break
-            except:
-                pass
-        # Only take the frames before the broken ones
-        ll = ll[:end+1]
+    reader = FFMpegVideoReader(filepath, normalize=normalize)
+    ll = LazyList.init_from_index_callable(reader.__getitem__, len(reader))
+    ll.fps = reader.fps
 
     return ll
+
+
+def ffmpeg_types():
+    r"""The supported FFMPEG types.
+
+    Returns
+    -------
+    supported_types : `dict`
+        A dictionary of extensions supported by the FFMPEG importer.
+    """
+    # Need to check this
+    ffmpeg_exts = ['.avi', '.mp4', '.mpg', '.mpeg', '.wmv', '.mov']
```

Comment:
I would definitely extend that to include more. E.g. mkv is quite common. 
You can find the supported ones with ffmpeg -formats.